mobile/CmpE451_App/src/AppNavigator.js

```diff
@@ -135,72 +145,116 @@ function BottomNavigator() {
     </Tab.Navigator>
   );
 }
+const startReload = ()=> RNRestart.Restart();
 
-function CustomDrawerContent(props) {
-  return (
-    <DrawerContentScrollView {...props}>
-      <FakeDrawerHeader>
-        <Image
-          source={{
-            uri: 'https://reactnative.dev/docs/assets/p_cat2.png',
-          }}
-          style={{width: 100, height: 100}}
-        />
-        <AppTitle> BOXY </AppTitle>
-      </FakeDrawerHeader>
-      <DrawerItemList {...props} />
-    </DrawerContentScrollView>
-  );
+const handleLogout = () => {
+  if (CONFIG.skipLogout) {
+    AsyncStorage.removeItem(KEYS.TOKEN_KEY);
+    startReload()
+  } else {
+    requestLogout()
+  }
+};
+
+const requestLogout = () => {
+      AXIOS_CLIENT.delete('logout')
+      .then(response => {
+        if (response.status === 200) {
+          AsyncStorage.removeItem(KEYS.TOKEN_KEY);
+          startReload()
+        } else {
+          ToastAndroid.show(TEXT.unexpectedError, ToastAndroid.SHORT);
+        }
+      })
+      .catch(error => {
+        console.info(error);
+        ToastAndroid.show(TEXT.networkError, ToastAndroid.SHORT);
+      });
 }
 
-const DrawerNavigator = () => {
-  return (
-    <Drawer.Navigator
-      screenOptions={screenOptionStyle}
-      drawerStyle={{backgroundColor: 'transparent'}}
-      drawerType={'slide'}
-      overlayColor="transparent"
-      initialRouteName="Navigator"
-      drawerContent={props => {
-        return <CustomDrawerContent {...props} />;
-      }}>
-      <Drawer.Screen
-        name="Navigator"
-        component={Navigator}
-        options={{
-          drawerItemStyle: {height: 0},
-          headerShown: false,
+const showAlert = () => {
+  Alert.alert(
+    'Warning',
+    'Are you sure you want to logout?', [
+  {
+    text: 'Cancel'
+  },
+  {
+    text: 'Yes',
+    onPress: () => handleLogout()
+  },
+    ],
+  {
+    cancelable: true,
+  })
+};
```

Comment:
We may put logout functionality in another file to improve readability. But this implementation also works! 🌟